### DISCUSSION TITLE
Module Federation: Only build container entry and exposed modules and not main.js

### DISCUSSION THREAD

**Discussion by Ka0o0** [webpack/webpack#16445] [Q&A]
Hi,

I'm working on a micro frontend architectured application and I was wondering if it was possible to only build the exposed modules and the remote container entry.

The background of this is that we have a lot of different components (each component is a build/container) that each export a certain number of modules. Those containers are not meant to be run "alone" but always within the Base container so they are always referenced as remotes from the Base.

Here is an example Webpack Config of the `dashboard` component which exposes a module `Dashboard` with the same name:

```TypeScript
import * as path from 'path';
import * as webpack from 'webpack';
import { VueLoaderPlugin } from 'vue-loader';

const deps = require('./package.json').dependencies;

const config: webpack.Configuration = {
  entry: "./src/index.ts",
  resolve: {
    extensions: [".tsx", ".ts", ".jsx", ".js", ".json"],
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: {
          loader: 'ts-loader',
          options: {
            configFile: path.resolve(__dirname, "./tsconfig.json"),
            appendTsSuffixTo: [/\.vue$/],
          }
        },
        exclude: /node_modules/,
      },
      {
        test: /\.m?js/,
        type: "javascript/auto",
        resolve: {
          fullySpecified: false,
        },
      },
      {
        test: /\.vue$/,
        loader: "vue-loader",
      },
      {
        test: /\.css$/i,
        use: ["style-loader", "css-loader"],
      },
      {
        test: /\.(js|jsx)$/,
        exclude: /node_modules/,
        use: "babel-loader",
      }
    ],
  },
  plugins: [
    new VueLoaderPlugin(),
    new webpack.container.ModuleFederationPlugin({
      name: "dashboard",
      filename: "remoteEntry.js",
      remotes: {},
      exposes: {
        './Dashboard': './src/Dashboard.vue',
      },
      shared: {
        "vue": {
          requiredVersion: deps["vue"]
        },
        "vue-router": {
          requiredVersion: deps["vue-router"]
        },
      },
    }),
  ],
};

export default config;
```

I'm building the component using `webpack --mode production --config ./webpack.config.ts`.
Now webpack produces the following files:

![image](https://user-images.githubusercontent.com/4035371/200262568-2fa747da-c173-4a2c-a8b7-d8f5c2d4b68b.png)

When looking into `remoteEntry.js` I can see that `66.js` and `393.js` are referenced but `341.js` is not.
On the other hand `main.js` references `341.js` and `393.js` but not `66.js`.
I don't quite understand the difference between `66.js` and `341.js` but it seems that the later is the  "non-module-federation" module.
Both of them seem to do the same thing.

From my understanding, the numbered files are the modules and main.js is the "non-module-federation" entrypoint of the build where modules are evaluated directly. 
Because we are only using the components via module federation remotes, I think we don't need all those non-module-federation files (`main.js` & `341.js`) and it would be superfluous to deploy them. 
Is there a way to  not generate them in the first place?

Thanks

**Comment by jedwards1211** [comment]
I would like a way to do this as well.  For now I'm working around it by using an empty `index.js` and ignoring the output `main.js`.

**Comment by jedwards1211** [comment]
Another workaround I've found, since I'm using the Webpack compiler API, is to just pass a nonexistent file for `entry` and ignore the error in the output `Compilation`.

**Comment by JasonKleban** [comment]
Related question: https://github.com/module-federation/core/discussions/2385

**Comment by JasonKleban** [comment]
How would you configure the generation of hashed names for 66.js and 393.js bundles but not for remoteEntry.js?
